text/0000-offline-buildpackages.md

```diff
@@ -0,0 +1,463 @@
+# Meta
+[meta]: #meta
+- Name: Asset Cache
+- Start Date: 2020-04-27
+- Author(s): dwillist
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: NA
+
+# Summary
+[summary]: #summary
+
+This change allows buildpacks to access cached assets during builds to enable offline builds and asset reuse.
+
+To do so we introduce the asset cache, a distribution artifact that provides vendored assets for buildpacks, and platform mechanisms to provide vendored assets to buildpacks during `build` and `detect`. We also add a top level `[[assets]]` array in `buildpack.toml`
+
+# Motivation
+[motivation]: #motivation
+
+- Simplify artifacts needed to achieve offline builds.
+- Enable registry deduplication of buildpack assets
+- Separate buildpacks from vendored assets.
+
+# What it is
+[what-it-is]: #what-it-is
+
+### Define the target persona:
+Buildpack user, Platform operator, Buildpack author
+
+### Explaining the feature largely in terms of examples.
+
+### Asset Cache Definition
+We define the asset cache artifact. It is a reproducible OCI image. Each layer in this image contains one or more `asset` files a buildpack may contribute during a future build. Each of these `asset` files will be made available at `/cnb/assets/<asset-sha256>` in the build container.
+
+##### General Layer Structure:
+```
+<layer1> ┳━ /cnb/assets/<java11-asset-sha256>
+         ┗━ /cnb/assets/<java13-asset-sha256>
+
+...
+<layern> ━━ /cnb/assets/<java15-asset-sha256>
+```
+
+### Asset Cache Image Labels
+Asset caches will have two labels. An`io.buildpacks.asset.cache.metadata` label that contains a human readable name.
+``` json
+{
+  "name": "asset-org/asset-name"
+}
```

Comment:
```suggestion
{
  "name": "asset-org/asset-name",
  "buildpack": {
    "id": "(required)",
    "version": "(required)"
  }
}
```
Given in the proposed workflows the asset cache is for a specific buildpack, does it make sense to the buildpack coordinates (id and version) to the metadata here?

Reply:
Also, if we tied the cache to a buildpack id/version do we still want or need a `name`? When would the name be useful?

Reply:
I had envisioned cache-images as having little info about how they are made or who is going to consume them, but I guess the only way we generate them now is with a `buildpack uri`.